### PR TITLE
Add the Logo component

### PR DESCRIPTION
**Pull request by ethanjc**
This PR add the logo component that supports various states with one or two text or React Node inputs.

CleanShot 2022-08-03 at 11 14 30
CleanShot 2022-08-03 at 11 15 32
CleanShot 2022-08-03 at 11 15 56
CleanShot 2022-08-03 at 11 16 17

### CODE REVIEW

File: packages/constellation/src/components/Compositions/Logo/Logo.tsx
Comment (noahub) [inline packages/constellation/src/components/Compositions/Logo/Logo.tsx]: It seems like there may be a need for transparency with images as primary logos. Maybe I'm misunderstanding what these mocks are getting at: https://www.figma.com/file/lI692dBvIP9Hmx2uCH0cnI/03---Web-%7C-Shared?node-id=4404%3A19240

Is there a way to account for that here?

Comment (ethanjc) [reply packages/constellation/src/components/Compositions/Logo/Logo.tsx]: Ah I thought that was just the image in the background but a transparent background makes sense. I can change it to only have a solid background when the value is text. Or the other option would be to just add a background colour prop.

Thoughts?

Comment (noahub) [reply packages/constellation/src/components/Compositions/Logo/Logo.tsx]: Hmm yea, an optional background color prop might be handy here. Might be the best option without knowing all the use cases upfront.

File: packages/constellation/src/components/Compositions/Logo/Logo.stories.tsx
Comment (noahub) [inline packages/constellation/src/components/Compositions/Logo/Logo.stories.tsx]: It feels like requiring that the consumer pass a whole image element or other react node could result in some potential misuse/misadventures.

Have you considered using `imgSrc` and `fallbackText` props instead for the logo image and text? That way we can better control the styling of the image?

Although this may depend on Blockchain.com's specific requirements. @blockdylanb are you aware of any specific requirements for how to pass images to these components?

Comment (ethanjc) [reply packages/constellation/src/components/Compositions/Logo/Logo.stories.tsx]: Yeah you have a very good point, possibly a safer option would be an imgSrc and I should add icons an as option there too

File: packages/constellation/package.json
Comment (noahub) [inline packages/constellation/package.json]: @ethanjc I think we should pin these versions.

File: packages/constellation/src/components/Compositions/Logo/Logo.tsx
Comment (noahub) [inline packages/constellation/src/components/Compositions/Logo/Logo.tsx]: How would you feel about consolidating these props into `primaryIcon` and `secondaryIcon`, both of which could be objects typed something like:
```
interface LogoIcon {
  icon: ...
  imgAlt: string
  imgSrc: string
  imgText: string
  variant?:
}
```
I think it would make it slightly easier for the consumer to include/omit icons this way. What do you think?

File: packages/constellation/package.json
Comment (noahub) [inline packages/constellation/package.json]: This version should probably be pinned as well.

File: packages/constellation/src/components/Compositions/Logo/Logo.tsx
Comment (ethanjc) [reply packages/constellation/src/components/Compositions/Logo/Logo.tsx]: Great idea!

File: packages/constellation/src/components/Compositions/Logo/Logo.stories.tsx
Comment (noahub) [inline packages/constellation/src/components/Compositions/Logo/Logo.stories.tsx]: I realize this makes it easier to modify the different aspects of the Logo within Storybook, but do you think this someone might confuse these as the required props (as opposed to the actual `primaryContent` and `secondaryContent` props?

I wonder if there is a way to call this out in the story or in a separate intro?

**Review by noahub — DISMISSED** [review]
I think that the Story props may be a little confusing, but don't think it should block this PR if you want to handle that separately.

File: packages/constellation/src/components/Compositions/Logo/Logo.stories.tsx
Comment (noahub) [inline packages/constellation/src/components/Compositions/Logo/Logo.stories.tsx]: Could you rm `ComponentStory` before merging?

**Review by noahub — APPROVED** [review]

**Pull request merged**
